### ISSUE TITLE
[Feature Request]: Add the ability to add a play with storiesOf

### ISSUE THREAD

**Issue by subskribe-kevin** [storybookjs/storybook#20036]
### Is your feature request related to a problem? Please describe

We're trying to build a number of stories via configurations so that every major property for a story is tested.

We're doing this with storiesOf() and add() but this won't work with plays.

### Describe the solution you'd like

It should be possible to build a normal story, with a play, and add it with add()

### Describe alternatives you've considered

Did a heads down analysis of the API and it seems like this is not possible.

### Are you able to assist to bring the feature to reality?

yes, I can

### Additional context

I can provide feedback on the ticket and look at code and test things.

**Comment by mehtajp22** [comment]
➕1
I have a similar use case too. I created a story and added a play function to it and added this story using `storiesOf().add` but I still see a message `No interactions found`.

For now I've been able to test my interactions on these stories with a workaround though. The type ArgsStoryFn receives the storyContext that has the canvasElement so I could do this:

```
const myStory: Story = (_, { canvasElement }) {
   //.. write interactions / assertions, etc. here, e.g., `userEvent.click(within(canvasElement).getByText('foo'))`

  return {
      component: MyComponent,
      moduleMetadata: {
      ...
    },
    ...
  }
}

storiesOf('someKind', module).add('storyName', myStory);
```

These interactions run locally on Storybook as well as on the Storybook published on Chromatic as expected, i.e., when I manually click on the Story. However, on the Chromatic build page where all my stories are listed, I don't see any interaction test failures on stories added in the above way. In fact I don't even see an "Interaction" story type mentioned for these stories. I just see the "Visual" test.
